MvvmHelpers/Utils.cs

```diff
@@ -1,38 +1,54 @@
-﻿using System;
+﻿using MvvmHelpers.Exceptions;
+using System;
+using System.Reflection;
 using System.Threading.Tasks;
 
 namespace MvvmHelpers
 {
-    /// <summary>
-    /// Extension Utils
-    /// </summary>
-    public static class Utils
-    {
-        /// <summary>
-        /// Task extension to add a timeout.
-        /// </summary>
-        /// <returns>The task with timeout.</returns>
-        /// <param name="task">Task.</param>
-        /// <param name="timeoutInMilliseconds">Timeout duration in Milliseconds.</param>
-        /// <typeparam name="T">The 1st type parameter.</typeparam>
-        public async static Task<T> WithTimeout<T>(this Task<T> task, int timeoutInMilliseconds)
-        {
-            var retTask = await Task.WhenAny(task, Task.Delay(timeoutInMilliseconds))
-                .ConfigureAwait(false);
+	/// <summary>
+	/// Extension Utils
+	/// </summary>
+	public static class Utils
+	{
+		/// <summary>
+		/// Task extension to add a timeout.
+		/// </summary>
+		/// <returns>The task with timeout.</returns>
+		/// <param name="task">Task.</param>
+		/// <param name="timeoutInMilliseconds">Timeout duration in Milliseconds.</param>
+		/// <typeparam name="T">The 1st type parameter.</typeparam>
+		public async static Task<T> WithTimeout<T>(this Task<T> task, int timeoutInMilliseconds)
+		{
+			var retTask = await Task.WhenAny(task, Task.Delay(timeoutInMilliseconds))
+				.ConfigureAwait(false);
 
-            return retTask is Task<T> ? task.Result : default(T);
-        }
+			return retTask is Task<T> ? task.Result : default;
+		}
 
-        /// <summary>
-        /// Task extension to add a timeout.
-        /// </summary>
-        /// <returns>The task with timeout.</returns>
-        /// <param name="task">Task.</param>
-        /// <param name="timeout">Timeout Duration.</param>
-        /// <typeparam name="T">The 1st type parameter.</typeparam>
-        public static Task<T> WithTimeout<T>(this Task<T> task, TimeSpan timeout) =>
-            WithTimeout(task, (int)timeout.TotalMilliseconds);
-        
-    }
+		/// <summary>
+		/// Task extension to add a timeout.
+		/// </summary>
+		/// <returns>The task with timeout.</returns>
+		/// <param name="task">Task.</param>
+		/// <param name="timeout">Timeout Duration.</param>
+		/// <typeparam name="T">The 1st type parameter.</typeparam>
+		public static Task<T> WithTimeout<T>(this Task<T> task, TimeSpan timeout) =>
+			WithTimeout(task, (int)timeout.TotalMilliseconds);
+
+#pragma warning disable RECS0165 // Asynchronous methods should return a Task instead of void
+		public static async void SafeFireAndForgetAsync(this Task task, Action<Exception> onException = null, bool continueOnCapturedContext = false)
+#pragma warning restore RECS0165 // Asynchronous methods should return a Task instead of void
+		{
+			try
+			{
+				await task.ConfigureAwait(continueOnCapturedContext);
+			}
+			catch (Exception ex)
```

Comment:
Careful - this is dangerous because it catches every exception thrown by the `Task`. 

I recommend not catching the Exception if the user has not implemented `onException` because it means that they don't want to gracefully handle the exception. I like to think of `onException` as the way the user implements a try/catch for `FireAndForgetAsync`; if they haven't implemented `onException`, don't catch any exceptions; if they have, catch 'em (all 😜).

```csharp
try
{
    await task.ConfigureAwait(continueOnCapturedContext);
}
catch (Exception ex)
{
    if(onException is null)
        throw;

    onException.Invoke(ex);
}
```

Same code, but optimized using Catch filters as recommended by the great Bill Wagner: https://github.com/brminnick/AsyncAwaitBestPractices/pull/8
```csharp
try
{
    await task.ConfigureAwait(continueOnCapturedContext);
}
catch (Exception ex) when (ex != null)
{
    onException.Invoke(ex);
}
```